### DISCUSSION TITLE
[deploee] 硬件模型库贡献协议-个人贡献者

### DISCUSSION THREAD

**Discussion by tpoisonooo** [open-mmlab/mmdeploy#1563] [General]
https://platform.openmmlab.com/deploee

## 一、贡献说明
把自己的开发板放到硬件模型库设备列表里，给大家一起用，避免吃灰。
## 二、操作细节
![流程图 (1)](https://github.com/open-mmlab/mmdeploy/assets/7872421/0a5c3943-0e43-419a-902b-4792d95b212d)
关键操作解释：

1. 填写开发板信息和 2. 确认是否邮寄：考虑到技术可能性和实际负载，某些设备无法接入硬件模型库。因此需要贡献者提供 <npu/gpu/cpu 类型，操作系统版本，贡献者联系方式>。

3. 邮寄到上海
  - 邮寄使用顺丰保价到付，由 openmmlab 预算支付邮费
  - 保价的计价方式，参照固定资产折旧办法
>取设备核心板（不含外围配件）购买价，从购入开始，每满一年折损 1/4，第五年残值为 0
以爱芯派为例，
20230303 从 sipeed 渠道购入核心板，价格为 279。此后无论 sipeed 涨价/降价/下架商品，基础价格均为 279
20240302 残值为 279
20240303 的残值为 279 * 3 / 4 = 209
20250303 残值为 279 * 2 / 4 = 139
- 发快递前，请仔细检查以下内容
  - 设备可用，电源、SD 卡（若运行需要）都在
  - 提供购买截图
  - 拍照保价备用

4. 确认设备无误
openmmlab 收到后拍照，于 3 个工作日内确认是否接收
    - 若运送期间物理损坏，由顺丰保价赔偿
    - 若物理表面未见损伤，但无法开机，退回
    - 若缺失电源、SD 卡等运行所需零件，退回
    - 退回仍由 openmmlab 支付顺丰保价运费

5. 建立github issue、上架硬件模型库
openmmlab 确认接收后，记录来源
    - 设备放置在自制的嵌入式设备架
    - 建立 github issue，记录此当前存放位置和状态
    - 5 个工作日内，设备上架到“硬件模型库”，具体取决于环境配置难度
      - 设备名称默认包含贡献者 github 名称。特殊情况（如名称过长）特殊处理
      - 若软件环境实在无法配置成功，退回

6. 退回
设备建立对应 github issue 后，无论何种原因发生的退回，都将更新 issue 状态；
对于 12345 阶段发生的退回，由 openmmlab 支付顺丰保价运费。

所有动作的执行方和确认形式如下表：
| 动作编号 | 执行动作 | 执行方 | 确认形式 |
| :-: | :-: | :-: | :-: |
| 1 | 填写开发板的 <npu/gpu/cpu 类型，操作系统版本，联系方式> | 贡献者 | 在 github issue https://github.com/open-mmlab/mmdeploy/discussions/1563 中回复表格 |
| 2 | 确认设备寄出前可运行且符合平台需求 | mmdeploy | 明确答复以下格式中的一个 <接收，收件地址，邮箱地址> 或 <不接收，原因> |
| 3 | 准备 <设备照片，开机正常照片，购买证明>，邮寄 | 贡献者 | 邮件发送 <设备正常开机图片，开发板图片，快递单号，购买证明> |
| 4 | 确认设备无误 | mmdeploy | github 明确答复 |
| 5 | 建立标签、github issue、上架 | mmdeploy | github issue 建立，即为已开始上架 |
| 6 | 退回 | mmdeploy | github issue 更新快递单号 |

## 三、用户信息保护
此过程会在 issue 中，公开展示贡献者 github 账号，仅用于记录来源。
1. mmdeploy 团队将会采取合理的措施保护用户的个人信息，除法律法规规定的情形外，未经用户许可mmdeploy 团队不会向第三方公开、透露用户个人信息。具体使用、保存、共享和转让信息的方式，可参见[《「OpenMMLab」个人信息保护政策》](https://openmmlab.com/policy)
2. 用户在参与本贡献事宜时，需要填写一些必要的信息（如Github账号、手机号、联系地址等），请保持这些信息的真实、准确、合法、有效并注意及时更新，以便mmdeploy 团队向用户提供及时有效的帮助，或及时寄回开发板。
3. mmdeploy 团队将运用必要的安全技术和程序建立完善的管理制度来保护用户的个人信息，以免遭受未经授权的访问、使用或披露。
4. mmdeploy 团队非常重视对未成年人个人信息的保护。若用户为18周岁以下的未成年人，在参与本贡献事宜前，应事先取得家长或法定监护人的书面同意。

## 四、用户承诺
1. 用户在参与本贡献事宜时须遵守法律法规，不得利用本活动从事以下行为，包括但不限于：
  (1)产品来源正当合法，非通过盗窃、侵占、抢夺、抢劫、诈骗、走私犯罪手段获得的二手数码产品或侵犯/可能侵犯合法商标权的假冒/仿冒二手数码产品；
  (2)非私自处分无权处分二手数码产品，尤其是依法查封/扣押/存在权利负担的二手数码产品；
  (3)其他法律法规禁止的行为。
2. 如果用户违反了本条约定，相关国家机关或机构可能会对用户提起诉讼、罚款或采取其他制裁措施，并要求 mmdeploy 团队给予协助。造成损害的，用户应依法予以赔偿，mmdeploy 团队不承担任何责任。
3. 用户违反本条约定，导致任何第三方损害的，用户应当独立承担责任；mmdeploy 团队因此遭受损失的，用户也应当一并赔偿。

## 五、维护说明
mmdeploy 接收设备并确认无误后
1. 收入 mmdeploy 自制的嵌入式设备架，统一管理
![图片](https://github.com/open-mmlab/mmdeploy/assets/7872421/7ce388df-1a35-49a4-a858-e1bf7e01b8fa)
2. 创建 github issue 记录此设备当前位置、获得途径，任何状态变更都会更新此 issue，供贡献方查看
3. 在 openmmlab 维护期间，出现“外伤”导致无法使用的，如
  - 主板破损；
  - 配件遗失；
由 openmmlab 尝试维修。无法维修的参照 “操作细节” 中的折旧办法定损赔偿。
4. 在维护期间，出现“内伤”导致无法使用的，如
  - 开发板自然老化；
  - 开机正常、运行时黑屏，然后再也也无法启动；
  - 配件（如 SD 卡）读写次数达到上限、电源异常短路、熔断器短路；
由 openmmlab 优先维修并通知用户，无法维修的寄回设备。

## 六、不可抗力及其他免责事由
1. 用户理解并同意，在本协议履行过程中，可能会遇到不可抗力等风险因素，使本贡献活动发生中断。不可抗力是指不能预见、不能克服并不能避免且对一方或双方造成重大影响的客观事件，包括但不限于如洪水、火灾、台风、地震、瘟疫流行、电力异常、风暴、战争、动乱、政府行为等。出现上述情况时，mmdeploy 团队将努力在第一时间与相关单位配合，及时进行修复，但是由此导致开发板的损坏，mmdeploy 团队无法赔偿用户。
2. 在法律允许的范围内，mmdeploy 团队对以下情形导致的用户损失不承担责任：
  (1)受到计算机病毒、木马或其他恶意程序、黑客攻击的破坏;
  (2)用户未依照 mmdeploy 团队列明的操作流程参与本活动的;
  (3)因设备维修留下的焊接痕迹；
  (4)其他 mmdeploy 团队无法控制或合理预见的情形。
  
## 七、管辖与法律适用
  本协议的成立、生效、履行、解释及纠纷解决，适用中华人民共和国大陆地区法律(不包括冲突法)。若用户和mmdeploy 团队之间发生任何纠纷或争议，首先应友好协商解决；协商不成的，用户同意将纠纷或争议提交上海市徐汇区有管辖权的人民法院管辖。

## 八、其他
  用户对本协议有任何疑问、意见、建议或投诉，请发送邮件至openmmlab@gmail.com与我们进行联系。在一般情况下，我们会在15个工作日内对您的请求予以答复。

**Comment by HePengguang** [comment]
赞
